util/tls/tls_socket.cc

```diff
@@ -165,6 +165,34 @@ auto TlsSocket::Close() -> error_code {
   return next_sock_->Close();
 }
 
+class SpinCounter {
+ public:
+  explicit SpinCounter(size_t limit) : limit_(limit) {
+  }
+
+  bool Check(bool condition) {
+    current_it_ = condition ? current_it_ + 1 : 0;
+    if (current_it_ >= limit_) {
+      current_it_ = 0;
+      return true;
+    }
+    return false;
+  }
+
+  size_t Limit() const {
+    return limit_;
+  }
+
+ private:
+  const size_t limit_;
+  size_t current_it_{0};
+};
+
+#define LOG_IF_SPIN_COUNT_LIMIT_REACHED(spin_count, res)                  \
+  if (spin_count.Check(res == 0)) {                                       \
```

Comment:
I didn't bother adding a `cond` instead of `res` since we only compare against `0` on all use cases so I didn't bother making it more generic nor did I bother to wrap it in `do/while(0)`

Reply:
Can you use `LOG_IF()` instead?